flink-core/src/main/java/org/apache/flink/api/common/functions/RuntimeContext.java

```diff
@@ -71,6 +71,17 @@
 	@PublicEvolving
 	MetricGroup getMetricGroup();
 
+	/**
+	 * Returned value is guaranteed to be unique between operators within the same job and to be
+	 * stable and the same across job submissions.
+	 *
+	 * <p>This operation is currently only supported in Streaming (DataStream) contexts.
+	 *
+	 * @return String representation of the operator's unique id.
+	 */
+	@PublicEvolving
+	String getOperatorUniqueID();
```

Comment:
Rather than adding this here and failing for *DataSet* programs, hoe about adding this to `StreamingRuntimeContext` and casting inside the Kafka Producer? Not super pretty, but nicer than having something that looks like a pretty generic concept (operator id) throwing an exception in a whole class of programs (batch jobs). This problem should go away anyways with the batch / streaming unification later.

Reply:
Probably better solution from the perspective of this particular use case, but do the UDF's have an access to `StreamingRuntimeContext`? If not, casting requirement would be huge blocker for any users from using this, thus making this almost a private API.

However I do not mind one way or the other.

PS, There are already operations/methods that do not work in streaming/batch

Reply:
I'm slightly leaning towards Stephan's suggestion, which I also agree is the better solution for this case.

It might be ok to have this as a "hidden" API for now anyways, since 1) it is marked `@PublicEvolving`, and 2) the API was added in quite a short timeframe.
If we want this fix in 1.5, I wouldn't suggest "fully" exposing it.

Reply:
Different point here, I think we would usually like to return `OperatorID` here and not string, but it is probably in a wrong module. When moving `OperatorID` seems to much, I would suggest to call the method something like `getOperatorIDAsString` to make this a bit more explicit.